### PR TITLE
create monitoring resources

### PR DESCRIPTION
**Pull request by pb82**
## Motivation

The keycloak operator should also create the resources required to enable monitoring for the keycloak instance.

## What

This PR adds phases to the operator for installing the monitoring resources (dashboard, service monitor, alerts). This might fail if the CRDs for those resources don't exist yet. In that case the operator will print a warning and retry.

## Verification Steps

1. Use a cluster that does not have any monitoring CRDs installed
1. Deploy keycloak using the operator (and this branch)
1. Keycloak should deploy fine but you should see warnings in the operator logs: `WARN[0060] Cannot create monitoring resource 'grafana-dashboard' at this time, retrying`
1. Now create the monitoring CRDs:
4.1 Grafana dashboard: https://github.com/integr8ly/grafana-operator/blob/master/deploy/crds/GrafanaDashboard.yaml
4.2 Prometheus rule: https://github.com/integr8ly/installation/blob/master/evals/roles/middleware_monitoring/templates/prometheus_rule_crd.yml.j2
4.3. Service monitor: https://github.com/integr8ly/installation/blob/master/evals/roles/middleware_monitoring/templates/prometheus_service_monitor_crd.yml.j2
1. The operator will now print out logs that creating the monitoring resources was successful.
1. Verify that those resources have been created.

You can also test it with the full installer by:

1. Updating the repo and branch to point to this PR here: https://github.com/integr8ly/installation/blob/master/evals/roles/rhsso/defaults/main.yml#L27
1. Updating the image to `pb82/keycloak-operator:monitoring` here: https://github.com/integr8ly/installation/blob/master/evals/roles/rhsso/defaults/main.yml#L2

## Progress

- [x] Finished task
- [ ] Verification

### CODE REVIEW

File: deploy/template/prometheus-rule.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            message: >-
+              {{ printf "%0.0f" $value }}% heap usage of {{ $labels.area }} in pod {{
+              $labels.pod }}, namespace {{ $labels.namespace }}.
+          expr: |
+            100 * jvm_memory_bytes_used{area="heap",namespace="[[ .Namespace ]]"}
+              / jvm_memory_bytes_max{area="heap",namespace="[[ .Namespace ]]"}
+              > 90
+          for: 1m
+          labels:
+            severity: warning
```
Comment (aidenkeating) [inline deploy/template/prometheus-rule.yaml]: @pb82 Based on this comment I think these should all be critical severity (https://github.com/integr8ly/installation/pull/333#issuecomment-461468700) ?

**Comment by aidenkeating** [comment]
@pb82 One comment, let me know when you're ready for a final verification

File: deploy/template/prometheus-rule.yaml (hunk 1 above)
Comment (pb82) [reply deploy/template/prometheus-rule.yaml]: good point, will update

**Comment by pb82** [comment]
@david-martin @philbrookes is this something we want to include in the upcoming release?

**Comment by david-martin** [comment]
@pb82 Yes. We'll have to write docs/SOPs for these alerts.
A follow up issue should suffice. That can be done outside the release process as they'll be stored in a separate repo

**Comment by pb82** [comment]
Alerts are created and the dashboard is imported after a full installer run:

keycloak_alerts

**Comment by pb82** [comment]
Tested the failed login attempts rule after a slight adjustment by clicking login (with invalid credentials) really often:

falied_logins

failed_logins_firing

**Pull request merged**